### PR TITLE
📷 Add camera object

### LINKED ISSUE #863: Add new Renderer.zoom method
Here is a method that can be used to zoom a plotter. We should add this to the `Renderer`, wrap it in `BasePlotter`, then fix the up/down key callbacks to use this as they are currently not triggering a render call (bug)

```py
import pyvista as pv

def zoom(plotter, value):
    if not plotter.camera_set:
        plotter.camera_position = plotter.get_default_cam_pos()
        plotter.reset_camera()
        plotter.camera_set = True
    plotter.camera.Zoom(value)
    plotter.render()
    
# Monkey patch it
pv.Plotter.zoom = zoom
```

And here is an exampple of it in use:
```py
from pyvista import examples

mesh = examples.load_airplane()

p = pv.Plotter()
p.add_mesh(mesh)
p.zoom(5)
p.show()
```

download

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add camera zoom and up

 
resolve #863

### Details
```Python
import pyvista as pv
from pyvista import examples

mesh = examples.load_airplane()
p = pv.Plotter()
p.add_mesh(mesh)
```
p

```Python
p.camera.zoom(5)
```
p2

```Python
p.camera.up([0, 1, 0])
```
p3

### CODE REVIEW

File: pyvista/plotting/renderer.py [hunk 1]
```diff
@@ -178,7 +179,16 @@ def camera_position(self, camera_location):
...
-        return self.GetActiveCamera()
+        camera = self.GetActiveCamera()
+        def zoom(self, value):
+            """Zoom of the camera."""
+            self.Zoom(value)
+        camera.zoom = types.MethodType(zoom, camera)
+        def up(self, vector):
+            """Up of the camera."""
+            self.SetViewUp(vector)
+        camera.up = types.MethodType(up, camera)
```
Comment (banesullivan) [inline pyvista/plotting/renderer.py:190]: We already have a method on the renderer and plotter to set the view up vector... I'm not sure what's the best way to proceed here.

This monkey path is fine by me (especially for zoom) but perhaps we need to create a new `Camera` class that wraps the camera with all these methods and then move those off the renderer with a deprecation warning

Comment (tkoyama010) [reply pyvista/plotting/renderer.py:190]: Thanks. I am +1 for new `Camera` class. I will consider the design of the new class.

Comment (banesullivan) [reply pyvista/plotting/renderer.py:190]: >  I will consider the design of the new class

That would be really awesome, thanks @tkoyama010!

**Comment by tkoyama010** [comment]
I am reading :memo: this to know how to use :camera: camera object in ```vtk```.
https://lorensen.github.io/VTKExamples/site/Python/GeometricObjects/Planes/

**Comment by tkoyama010** [comment]
@akaszynski @banesullivan Now it is ready for review but I have 1 question. I think it is better to integrate `CameraPosition` class to `Camera` class. How do you think?

**Comment by tkoyama010** [comment]
Oops I had error in CI. Please wait...

File: pyvista/plotting/renderer.py [hunk 2]
```diff
@@ -96,6 +97,133 @@ def viewup(self, value):
         self._viewup = value
 
 
+class Camera:
```
Comment (akaszynski) [inline pyvista/plotting/renderer.py:100]: We really should be subclassing here.

```py
class Camera(vtk.vtkCamera):
```

This follows the same approach we've used for the other classes (i.e. ``pyvista.Renderer`` so we can still pass objects of this class to vtk, and anyone wanting to use this class in a legacy manner still can.

Comment (tkoyama010) [reply pyvista/plotting/renderer.py]: I fixed it. Thanks for your comment :+1:

**Comment by akaszynski** [comment]
Nice job subclassing `vtkCamera`!

Still two items remaining:

- Move the `Camera` class to its own file.  There's a good discussion on SO [here](https://stackoverflow.com/a/107836) talking about modules and classes.  While there's no specific standard (PEP8 or otherwise), I think based on our current layout, we should move Camera outside of `renderers.py` and into its own module named `camera.py`.

- Improve coverage with unit testing specific to the camera class (see [coverage](https://app.codecov.io/gh/pyvista/pyvista/compare/945/tree/pyvista/plotting/renderer.py)).  We should have a new `test_camera.py` within `tests`

**Comment by tkoyama010** [comment]
Thanks @akaszynski it is a nice idea. I'll add that.

**Comment by tkoyama010** [comment]
@akaszynski @banesullivan It is ready for review.
(CI in macOS Python36 is fail in install pyvista. I cannot understand why it is fail :confused: )

**Review by akaszynski — APPROVED** [review]
@tkoyama010, I think this is great.  I'd like to merge this PR and then create another that adds additional documentation and examples for controlling the camera.

Thanks for your hard work, and sorry for not reviewing it earlier.

**Pull request merged**